app/models/course/assessment/answer/programming_file.rb

```diff
@@ -10,6 +10,22 @@ class Course::Assessment::Answer::ProgrammingFile < ActiveRecord::Base
   has_many :annotations, class_name: Course::Assessment::Answer::ProgrammingFileAnnotation.name,
                          dependent: :destroy, foreign_key: :file_id, inverse_of: :file
 
+  # Returns the code from start line to end line.
+  #
+  # @param [Integer] line_start 1 based line number (first line is 1).
+  # @param [Integer] line_end 1 based line number (first line is 1).
+  # @return [String|nil] the code in the lines specified. `nil` will be returned if the input is not
+  #   valid.
+  def code_at_lines(line_start, line_end)
```

Comment:
@lowjoel I tried this, then we will have too much computation in views:

```
code = file_annotation.file.code_at_lines(file_annotation.line - 5, file_annotation.line)
```

will become:
```
 start_line = file_annotation.line - 5
 start_line = 1 if start_line < 1
 end_line = file_annotation.line
 code = file.lines[(start_line - 1)..(end_line - 1)].join("\n")
```

In this case, I would still lean to let the model handle the computation, what do you think ?